Update Command Regex for Email Command And Add Optional Timestamp Support

zkfriendly opened the pull request:
## Description
This PR adds support for optional timestamp handling in email authentication and updates the command regex pattern. The system now works with both timestamped and non-timestamped emails while keeping existing functionality intact.

Key changes:
- Added optional timestamp handling: outputs 0 for timestamp values in the no-timestamp version
- Updated command regex pattern to `(<div id=3D\"[^\"]*zkemail[^\"]*\"[^>]*>)`
   - removes the requirement to immediately see `</div>` at the end of command
   - this relaxes constraints and allows for more email support like iCloud that adds an additional `<br>` tag before `</div>`
- Maintained support for existing email formats and providers

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
The changes have been tested with different email providers and formats:

- [x] Outlook emails with English settings on PC
- [x] Gmail emails sent from Apple Mail
- [x] iCloud emails with English settings on PC
- [x] Verified timestamp handling (outputs 0 for both timestamped and non-timestamped emails)
- [x] Tested command regex pattern matching across different email formats

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

benceharomi commented on bun.lock:
I thought you already added `bun.lock` to `.gitignore`

zkfriendly replied:
I see. Maybe it was some other repo.

benceharomi replied:
can you add it (and remove the committed bun.lock) here too?

JohnGuilding commented:
@zkfriendly I won't review these changes, but can you provide an explanation of the CI changes you made and how they fixed things? Would be interested

zkfriendly commented:
> @zkfriendly I won't review these changes, but can you provide an explanation of the CI changes you made and how they fixed things? Would be interested

major changes:
 - refactored and published nightly packages for `relayer-utils` to make it compatible with jest
 - changed tests to use jest rather than bun which has issues with the wasm build of `relayer-utils`
 - use same github workflows as in the main branch
 - only use one active worker per test suit (to prevent jobs getting killed due to excess RAM usage)

benceharomi commented on packages/circuits/package.json:
```diff
@@ -4,23 +4,27 @@
   "version": "1.0.3",
   "scripts": {
     "build": "mkdir -p build && circom src/email_auth.circom --r1cs --wasm --sym --c -l ../../node_modules -o ./build",
+    "build-no-timestamp": "mkdir -p build/no-timestamp && circom src/email_auth_no_timestamp.circom --r1cs --wasm --sym --c -l ../../node_modules -o ./build/no-timestamp",
```
what about using an ENV var?

zkfriendly replied:
> question: Is there a better way to have these scripts than putting them here?

I mean ideally we could use a framework like circomkit to make our life's a lot easier and manage our circuits in a more structured way

> what about using an ENV var?
how do you mean?

benceharomi replied:
instead of having two separate contracts for with and without timestamp, having a boolean env var that can be checked to decide which circuit to use?

zkfriendly replied:
So we currently have 4 different scripts:

1. `build`
2. `build-no-timestamp`
3. `build-legacy`
4. `build-recipient`

To make sure I understand your suggestion: are you suggesting that we introduce a boolean ENV variable to handle case 1 and 2? and leave others as is? 

personally I'm not sure if this simplifies things since we would be introducing a different mechanism for two of the scripts, and also we would need to update documentation on how to use the ENV variable.

Also making the scripts completely dynamic seems like we would be making a framework like `Circomkit`. In which case I wonder why we shouldn't use `Circomkit` (would need its own PR).

shreyas-londhe reviewed (approved):
LGTM
